packages/keyring-controller/src/KeyringController.ts

```diff
@@ -256,10 +257,14 @@ export type KeyringControllerOptions = {
  * Keyring object to return in fullUpdate
  * @property type - Keyring type
  * @property accounts - Associated accounts
+ * @property typeIndex - The index of the keyring in the array of keyrings of the same type
+ * @property id - The id of the keyring
  */
 export type KeyringObject = {
   accounts: string[];
   type: string;
+  typeIndex: number;
```

Comment:
I think that storing the this `typeIndex` value would make things unnecessarily complicated. Can't we just use the ID to get its index programmatically?

Given that we may have multiple HDKeyrings that are restored with the same order every time the wallet is unlocked (the keyring instances are recreated), if one in the middle of the array would get removed we would have to reassign `typeIndex` to all other keyrings of the same type, which would make the cost / benefit of this unclear to me